src/main/java/uk/gov/justice/hmpps/prison/api/resource/OffenderResource.java

```diff
@@ -201,4 +202,21 @@ CaseNote updateOffenderCaseNote(@ApiParam(value = "Noms ID or Prisoner number (a
     PrivilegeSummary getLatestBookingIEPSummaryForOffender(
             @ApiParam(value = "offenderNo", required = true, example = "A1234AA") @PathVariable("offenderNo") @NotNull String offenderNo,
             @ApiParam(value = "Toggle to return IEP detail entries in response (or not).", required = true) @RequestParam(value = "withDetails", required = false, defaultValue = "false") boolean withDetails);
+
+    String NOMS_ID_REGEX_PATTERN = "[a-zA-Z][0-9]{4}[a-zA-Z]{2}";
+
+    @SuppressWarnings("RestParamTypeInspection")
+    @GetMapping("/{offenderNo}/transaction-history")
+    @ApiOperation(value = "Retrieve an offender's financial transaction history for cash, spends or savings.",
+            notes = "Transactions are returned in NOMIS order (Descending date followed by id).<br/>" +
+            "All transaction amounts are represented as pence values.") //todo order?, should use int for money?
+    @ApiResponses(value = {
+            @ApiResponse(code = 400, message = "Not a digital offender. Offender not found. Offender has no account at this prison.", response = ErrorResponse.class),
+            @ApiResponse(code = 404, message = "Prison, offender or accountType not found", response = ErrorResponse.class),
+            @ApiResponse(code = 500, message = "Unrecoverable error occurred whilst processing request.", response = ErrorResponse.class)})
+    ResponseEntity<List<OffenderTransactionHistoryDto>> getTransactionsHistory(
+            @ApiParam(name = "offenderNo", value = "Offender ID", example = "123", required = true) @PathVariable(value = "offenderNo", required = true) @NotNull Long offenderId,
+            @ApiParam(name = "account_code", value = "Account code", example = "spends", required = false, allowableValues = "spends,cash,savings") @RequestParam(value = "account_code", required = false) String accountCode,
+            @ApiParam(name = "from_date", value = "Start date for transactions, format yyyy-MM-dd, defaults to today if not supplied", example = "2019-04-01") @DateTimeFormat(iso = DateTimeFormat.ISO.DATE) @RequestParam(value = "from_date", required = false) LocalDate fromDate,
```

Comment:
I would use String instead of LocalDate for dates so that a more precise message is provided when a bad format is used

Reply:
Hmmm a noble aim but I'm not sure it's a good enough reason to lose the typing on the API.  Maybe there's another way to improve the error message?  Happy to assist.

Reply:
done